### PR TITLE
Add resque plus prepare and perform transfer jobs

### PR DESCRIPTION
**Pull request by elohanlon**

### CODE REVIEW

File: app/validators/source_object_checksum_validator.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Ensures that this object's associated source_object has been assigned a fixity_checksum_value.
+class SourceObjectChecksumValidator < ActiveModel::Validator
+  def validate(record)
+    return if record.source_object&.fixity_checksum_value.present?
```
Comment (barmintor) [inline app/validators/source_object_checksum_validator.rb:6]:
```suggestion
    return if (record.source_object&.fixity_checksum_value.present? && record.source_object&.fixity_checksum_algorithm_id.present? )
```

File: config/templates/aws.template.yml [hunk 2]
```diff
@@ -6,9 +6,13 @@ development:
   preservation_bucket_name: 'example-dev-bucket'
   aws_access_key_id: 'access_key_id'
   aws_secret_access_key: 'secret_access_key'
+  local_path_key_map:
```
Comment (barmintor) [inline config/templates/aws.template.yml:9]:
```suggestion
  # the local path segment should have a trailing slash to ensure that object keys do not begin with a slash
  local_path_key_map:
```

File: app/validators/source_object_checksum_validator.rb (hunk 1 above)
Comment (elohanlon) [reply app/validators/source_object_checksum_validator.rb:6]: Will update in upcoming commit! (plus separate validation messages for missing value vs. algorithm)

**Review by barmintor — APPROVED** [review]
:shipit:

**Pull request merged**